dig.go

```diff
@@ -648,12 +698,14 @@ func shallowCheckDependencies(c containerStore, p param) error {
 
 		if ns := c.getValueProviders(ps.Name, ps.Type); len(ns) == 0 && !ps.Optional {
 			missing = append(missing, newErrMissingType(c, key{name: ps.Name, t: ps.Type}))
+			missingNodes = append(missingNodes, ps.DotParam()...)
 		}
 
 		return true
 	}))
 
 	if len(missing) > 0 {
+		c.missingNodes(missingNodes)
```

Comment:
It might be better to update the name to `addMissingNodes` so that it's more clear that there is some mutation happening here.

Reply:
Updated to `addMissingNodes`.